File: Social_medie_projekt/WebApi/DTOs/SignInResponse.cs
namespace WebApi.DTOs
{
    public class SignInResponse
    {
        public int LoginId { get; set; }

        public string Email { get; set; } = string.Empty;

        public Role Role { get; set; }

        public SignInUserResponse? User { get; set; }

        public string? Token { get; set; }
    }

    public class SignInUserResponse
    {
        public int UserId { get; set; }

        public string UserName { get; set; } = string.Empty;

        public DateTime Created { get; set; }

        public SignInUserUserImageResponse UserImage { get; set; } = new();
    }

    public class SignInUserUserImageResponse
    {
        public string? Image { get; set; }
    }
}

File: Social_medie_projekt/Client/src/app/Req-Res/signInResponse.ts
import { Role } from "../_models/role";

export interface SignInResponse {
    token: string;
    loginResponse: LoginResponse
}

export interface LoginResponse {
    loginId: number;
    email: string;
    type: Role;
    user: userResponse;
}

export interface userResponse {
    userId: number;
    userName: string;
    created: Date;
    posts: postResponse;
}

export interface postResponse{
    postId: number;
    userId: number;
    title: string;
    desc?: string;
    tags?: string;
    pictureURL?: string;
    likes: number;
    date: Date;
}